### PR TITLE
Feat: Refactors section grid

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
This PR is a follow up from:
- https://github.com/vtex/faststore/pull/1775
- https://github.com/vtex/faststore/pull/1768

Doing these other tasks, we found that it wasn't possible to change a section from full width to grid centered, so we fixed it in order to improve the dev experience using the FS 2.0.

Additionally, we:
- Set `ProductCard` bordered as default
- Removed old `.layout__main` class
- Improved `Carousel` padding to always follow the centered grid
- Updated customization docs naming and descriptions (view @Mariana-Caetano commits after [comment](https://github.com/vtex/faststore/pull/1779#pullrequestreview-1475962544))

## How it works?

https://github.com/vtex/faststore/assets/11613011/e3f17c79-b6f9-4a06-9e60-65df0b806954

## How to test it?
All components should look the same as `main`.

`library` doc:
1. Review new doc [Customizing Pages](https://faststore-site-git-feat-section-grid-fs-1074-faststore.vercel.app/docs/customizing/pages)

`faststore/core` application:
1. Run `yarn dev`
2. Check all pages

`starter.store`:
1. Visit our [preview link](https://evergreen-git-feat-theming-test-faststore.vercel.app/)

### CODE REVIEW

**Review by eduardoformiga — COMMENTED** [review]
Good job, @renatamottam  you rock!

I left some comments regarding the additional `data-fs-content-*` attributes and wrapper nodes, can we use the already existing ones? 
Would be also good to validate this changes in the `starter.store` before merging for the sake of LH performance, since I think we are almost in the limit, and this change is kinda big.

File: packages/core/src/styles/global/utilities.scss
Comment (eduardoformiga) [inline packages/core/src/styles/global/utilities.scss:1]: hey, we don't need this duplicated file anymore, just the one from @faststore/ui. Take a look at 
https://github.com/vtex/faststore/pull/1792

File: packages/components/src/molecules/NavbarLinks/NavbarLinks.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ const NavbarLinks = forwardRef<HTMLDivElement, NavbarLinksProps>(
   ) {
     return (
       <nav data-fs-navbar-links ref={ref} data-testid={testId} {...otherProps}>
-        {children}
+        <div data-fs-navbar-links-wrapper>{children}</div>
```
[the remark this answers is not in the document]
Comment (renatamottam) [reply packages/components/src/molecules/NavbarLinks/NavbarLinks.tsx:18]: we do need this :(

File: apps/site/pages/docs/_meta.json [hunk 2]
```diff
@@ -17,7 +17,13 @@
     }
   },
   "customizing": {
-    "title": "Customizing",
+    "title": "Customizing Components",
```
Comment (hellofanny) [inline apps/site/pages/docs/_meta.json:20]: I think we can create a sub page under Customizing with `Customizing Pages`. WDYT?

File: apps/site/pages/docs/customizing-pages.mdx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+<header className="hero">
+
+# Customizing Pages
+
+We provide a couple mixins to help you styles your page sections.
+
+</header>
+
+## Styling a section
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing-pages.mdx:21]: We probably need to explain what it's a page section here.
I hope we can have this explanation in this [page](https://www.faststore.dev/docs/reference-guides/list-of-available-native-sections) soon.

File: apps/site/pages/docs/customizing-pages.mdx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+      }
+    }
+    ```
+
+  </Tab>
+</Tabs>
+
+### Full width layout
+
+If for some reason you want to use your section's **content with full width layout** you'll need to change it's content to include the `@mixin layout-content-full`.
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing-pages.mdx:213]:
```suggestion
If for some reason you want to use your section's content with  **full width layout** you'll need to change it's content to include the `@mixin layout-content-full`.
```

**Comment by hellofanny** [comment]
@Mariana-Caetano Mari can you please review this doc [page](https://faststore-site-git-feat-section-grid-fs-1074-faststore.vercel.app/docs/customizing/pages) related to customizing pages? Thanks : 😊

File: apps/site/pages/docs/customizing/pages.mdx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+    ```scss
+    // This is already set as default
+    [data-fs-content] { @include layout-content; }
+    ```
+  </Tab>
+</Tabs>
+
+### Content centered + Colored bg
+
+If you want to add a **colored background** with **full width** and mantain the **content centered**, you should add `@include layout-colored` to your desired section.
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/customizing/pages.mdx:125]:
```suggestion
To display a *colored background* with *full width* and maintain the *content centered*, add `@include layout-colored` to your desired section.
```

File: apps/site/pages/docs/customizing/pages.mdx [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+      }
+    }
+    ```
+
+  </Tab>
+</Tabs>
+
+### Content centered + Colored bg + Section centered
+
+If you want to add a **colored background** and **centering** the `.section`, you should add `@include layout-colored` and `@include layout-content` to your desired section.
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/customizing/pages.mdx:168]:
```suggestion
To add a *colored background* and *centering* the `.section`, add `@include layout-colored` and `@include layout-content` to your desired section.
```

File: apps/site/pages/docs/customizing/pages.mdx [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+      }
+    }
+    ```
+
+  </Tab>
+</Tabs>
+
+### Full width layout
+
+If for some reason you want to use your section's content with **full width layout** you'll need to change it's content to include the `@mixin layout-content-full`.
```
Comment (Mariana-Caetano) [inline apps/site/pages/docs/customizing/pages.mdx:213]:
```suggestion
To use your store section's content with a *full-width layout* you'll need to change its content to include the `@mixin layout-content-full`.
```

**Review by Mariana-Caetano — COMMENTED** [review]
Awesome @renatamottam ✨
I left some suggestions, and I have one specific for the docs titles:
Our sidebar has the` Customizing` section in `theming`, `Customizing Components` within this section, and `Component Customization` in `How to guides`. This can cause some confusion at first sight.

Perhaps we can change the name of the `Customizing` section in `theming` to `Customizing with Design Tokens`.

img-issue

File: packages/ui/src/styles/base/layout.scss [hunk 8]
```diff
@@ -11,39 +11,32 @@ body {
...
+  flex-direction: column;
+}
+
 .layout__main {
   display: flex;
   flex-wrap: wrap;
   background: var(--fs-body-bkg);
 }
 
 .layout__section {
```
[the remark this answers is not in the document]
Comment (renatamottam) [reply packages/ui/src/styles/base/layout.scss:17]: we can do this in another task :)

**Review by icazevedo — APPROVED** [review]
LGTM

File: packages/core/src/components/product/ProductCard/ProductCard.tsx [hunk 9]
```diff
@@ -55,7 +55,7 @@ export interface ProductCardProps {
 function ProductCard({
   product,
   index,
-  bordered = false,
+  bordered = true,
```
Comment (icazevedo) [inline packages/core/src/components/product/ProductCard/ProductCard.tsx:58]: Why are we changing this default?

Comment (renatamottam) [reply packages/core/src/components/product/ProductCard/ProductCard.tsx:58]: It has a more neutral design and don’t make us do fixes on the page layout to fits it. Example:

image

**Pull request merged**